drivers/wifi/esp32/Kconfig.esp32

```diff
@@ -5,6 +5,10 @@ menuconfig WIFI_ESP32
 	depends on !SMP
 	select THREAD_CUSTOM_DATA
 	select WIFI_USE_NATIVE_NETWORKING
+	select MBEDTLS
+	select MBEDTLS_ENTROPY_ENABLED
+	select MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+	select MBEDTLS_ECP_ALL_ENABLED
```

Comment:
Really, all ECP are needed?

Reply:
Also: could you clarify in commit message why are you pulling those mbedTLS dependencies? It is not clear to me.

Reply:
@mniestroj I will add this info into commit message. I just moved `MBEDTLS*` from `prj.conf` as this is mandatory dependency. https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/esp32/wifi_station/prj.conf#L17  
I'll work you ECP dependencies in another PR. For now, just keeping what is already upstreamed.